### PR TITLE
Added new API for exposing cache statistics in the service layer

### PR DESCRIPTION
**Pull request by jdrossl**
https://github.com/craftercms/craftercms/issues/1999

### CODE REVIEW

File: src/main/java/org/craftercms/core/cache/impl/store/MapCacheStoreAdapter.java [hunk 1]
```diff
@@ -147,6 +148,13 @@ public void clearScope(String scope) throws Exception {
         getScopeCache(scope).clear();
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    public CacheStatistics getStatistics(String scope) {
+        return CacheStatistics.EMPTY;
```
Comment (avasquez614) [inline src/main/java/org/craftercms/core/cache/impl/store/MapCacheStoreAdapter.java]: Can't we return the size in the case of MapCacheStoreAdapter?

File: src/main/java/org/craftercms/core/service/CacheService.java [hunk 2]
```diff
@@ -114,4 +115,9 @@ void put(Context context, Object key, Object value, List<Object> dependencyKeys,
      */
     void clearScope(Context context) throws InvalidContextException, InternalCacheEngineException;
 
+    /**
+     * Returns the statistics for all scopes of the given context.
```
Comment (avasquez614) [inline src/main/java/org/craftercms/core/service/CacheService.java]: Seems like the comment is wrong... it's a single scope per context.

**Comment by sumerjabri** [comment]
I'm freezing `commons` merges until we release 3.0.11. Please leave open.

File: src/main/java/org/craftercms/core/cache/Cache.java [hunk 3]
```diff
@@ -228,4 +228,11 @@ void put(String scope, Object key, Object value, List<Object> dependencyKeys, lo
      */
     void clearScope(String scope) throws InvalidScopeException, InternalCacheEngineException;
 
+    /**
+     * Returns the statistics for all scopes in the cache.
```
Comment (avasquez614) [inline src/main/java/org/craftercms/core/cache/Cache.java]: Sorry missed this one :). Fix it to by changing to `Returns the statistics of the specified scope...`

**Pull request merged**